### PR TITLE
docs: update dashboard-variable docs

### PR DESCRIPTION
**Pull request by gabor**
i adjusted the documentation to better reflect the situation with dashboard variables.

i thin it contains all the info we want to provide, but i think the phrasing could be improved .. somehow.

in general, this is the info i'm trying to provide:
- usually you don't need to think about all the `__text` and `__value` stuff. usually you just run a query, it returns some strings, grafana extracts all those strings and all is good.
- sometimes, you want to go one step further, and you want to have the dropdown-widget that offers the possible variable values contain special display-names for the given values.  some data sources allow this. the rules are different from datasource to datasources, there are two approach that are somewhat common

comments & changes very welcome :)

### CODE REVIEW

File: docs/sources/dashboards/variables/add-template-variables/index.md [hunk 1]
```diff
@@ -132,7 +132,8 @@ Query expressions can contain references to other variables and in effect create
...
 1. In the **Query** field, enter a query.
    - The query field varies according to your data source. Some data sources have custom query editors.
-   - Make sure that the query returns values named `__text` and `__value` as appropriate in your query syntax. For example, in SQL, you can use a query such as `SELECT hostname AS __text, id AS __value FROM MyTable`. Queries for other languages will vary depending on syntax.
+   - The exact way how the query variable data is extracted from the query results is specific to the given data source, please check the data source's documentation, but usually every string value returned from the data source is used as a possible variable value.
+     - Additionally, some data sources allow providing custom "display names" for the values, for example the PostgreSQL, MySQL and the Microsoft SQL Server plugins handle this by looking for fields named `__text` and `__value` in the result. Other data sources look for `text` and `value` or use a different approach. Again, please refer to the documentation for the specific data source plugin.
```
Comment (axelavargas) [inline docs/sources/dashboards/variables/add-template-variables/index.md:136]: 💡 Maybe something like:

```suggestion
- Each data source defines how the variable values are extracted. The common implementation is to use every string value returned from the data source response as a variable value. Make sure to double-check the data source documentation.
- Some data sources allow providing custom "display names" for the values. For instance, the PostgreSQL, MySQL, and Microsoft SQL Server plugins handle this by looking for fields named __text and __value in the result. Other data sources may look for `text` and `value` or use a different approach. Always remember to double-check the data source documentation.
```

**Review by axelavargas — APPROVED** [review]
Great work correcting the documentation @gabor 🙌🏾 , I've left a suggestion on how we can write this down, but from the technical accuracy is great.

File: docs/sources/dashboards/variables/add-template-variables/index.md [hunk 2]
```diff
@@ -131,9 +131,12 @@ Query expressions can contain references to other variables and in effect create
    - **On Dashboard Load:** Queries the data source every time the dashboard loads. This slows down dashboard loading, because the variable query needs to be completed before dashboard can be initialized.
    - **On Time Range Change:** Queries the data source every time the dashboard loads and when the dashboard time range changes. Use this option if your variable options query contains a time range filter or is dependent on the dashboard time range.
 1. In the **Query** field, enter a query.
+
    - The query field varies according to your data source. Some data sources have custom query editors.
-   - Make sure that the query returns values named `__text` and `__value` as appropriate in your query syntax. For example, in SQL, you can use a query such as `SELECT hostname AS __text, id AS __value FROM MyTable`. Queries for other languages will vary depending on syntax.
+   - Each data source defines how the variable values are extracted. The common implementation is to use every string value returned from the data source response as a variable value. Make sure to double-check the data source documentation.
```
Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:136]:
```suggestion
   - Each data source defines how the variable values are extracted. The typical implementation uses every string value returned from the data source response as a variable value. Make sure to double-check the documentation for the data source.
```

File: docs/sources/dashboards/variables/add-template-variables/index.md [hunk 3]
```diff
@@ -131,9 +131,12 @@ Query expressions can contain references to other variables and in effect create
...
+   - Some data sources allow providing custom "display names" for the values. For instance, the PostgreSQL, MySQL, and Microsoft SQL Server plugins handle this by looking for fields named `__text` and `__value` in the result. Other data sources may look for `text` and `value` or use a different approach. Always remember to double-check the data source documentation.
```
Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:137]:
```suggestion
   - Some data sources let you provide custom "display names" for the values. For instance, the PostgreSQL, MySQL, and Microsoft SQL Server plugins handle this by looking for fields named `__text` and `__value` in the result. Other data sources may look for `text` and `value` or use a different approach. Always remember to double-check the documentation for the data source.
```

**Review by imatwawana — APPROVED** [review]
Thanks so much for this update, @gabor! I've made a couple small wording suggestions to hit a couple points harder and make some of it more direct, but otherwise this LGTM!

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. Just letting you know I'm updating your PR's milestone to [11.2.x](https://github.com/grafana/grafana/milestone/529). If you need to see your changes in a previous version, please consider adding a backport label.

Thank you!

**Pull request merged**